content/en/blog/istio-traffic-management-walkthrough/index.md

```diff
@@ -0,0 +1,1162 @@
+        - destination:
+            host: greeter-service.default.svc.cluster.local
+            port:
+              number: 3000
+            subset: v2
+    - route:
+        - destination:
+            host: greeter-service.default.svc.cluster.local
+            port:
+              number: 3000
+            subset: v1
+EOF
+~~~~~~~~
+
+Don't forget to include the other route destination at the end, so any request that doesn't match the `/greeting` will be routed correctly as well. If you forget to add that last route, Istio won't know where to route the traffic if the match condition evaluates to false.
+
+Try deploying the above YAML and then open the `$GATEWAY/greeting` URL. When the request gets matched, the `/greeting` prefix is detected, the URL gets rewritten to `/hello` and it gets routed to the defined destination.
+
+Similarly, you could use an HTTP redirect as well:
+
+~~~~~~~~sh
+cat <<EOF | kubectl apply -f -
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: greeter-service
+spec:
+  hosts:
+    - greeter-service.default.svc.cluster.local
+  http:
+    - match:
+      - uri:
+          prefix: /greeting
+      redirect:
+         uri: /hello
+         authority: greeter-service.default.svc.cluster.local:3000
+    - route:
+        - destination:
+            host: greeter-service.default.svc.cluster.local
+            port:
+              number: 3000
+EOF
+~~~~~~~~
+
+Usually, you'd use the HTTP redirect if you want to redirect the call to a completely different service. Let's say we had another service called `better-greeter-service` and we tried to redirect traffic to that service instead. In that case, you can update the `authority` to `better-greeter-service` and the traffic will get redirected to that service.
+
+Just like you've defined a single match condition for one destination, we could also define multiple match conditions for each route OR add more conditions to a single match. If we build on the previous example, you can add additional match that would check if there's a specific value in the headers:
+
+~~~~~~~~sh
+...
+http:
+    - match:
+          - uri:
+                prefix: /greeting
+            headers:
+                x-my-header:
+                    exact: something
+      redirect:
+          uri: /hello
+          authority: greeter-service.default.svc.cluster.local:3000
+...
+~~~~~~~~
+
+In the above case we redirect requests to `/hello` endpoint if both match conditions are satisfied (`AND` semantics). Similarly, you could add an additional match entry:
+
+~~~~~~~~sh
+...
+http:
+  - match:
+    - uri:
+        prefix: /greeting
+    redirect:
+      uri: /hello
+      authority: greeter-service.default.svc.cluster.local:3000
+  - match:
+    - uri:
+      headers:
+        x-my-header:
+          exact: something
+    redirect:
+      uri: /blah
+      authority: greeter-service.default.svc.cluster.local:3000
+...
+~~~~~~~~
+
+## Dark Traffic (Mirroring)
```

Comment:
do we want to split the Dark Traffic into a separate blog post?

Reply:
This article is so long that we can divide it into two sections.